Updating setSolidFraction to be smooth and have an option for compressible and adding LPBF_small tutorial for the compressible solver

cosicp opened the pull request.

Commit:
updating setSolidFraction to be smooth and adding option for compressible

Commit:
adding compressible LPBF_small tutorial

Commit:
separating old and updated setSolidFraction for now so theres no problems in github test

philipcardiff commented on applications/utilities/compressibleSetSolidFraction/setSolidFraction.C:
```diff
@@ -0,0 +1,408 @@
+/*---------------------------------------------------------------------------*\
```
Rename this `setSolidFraction.C` to `compressibeSetSolidFraction.C` to avoid confusion

philipcardiff commented:
What's the difference between `setSolidFraction` and `compressibleSetSolidFraction`? Do you think we could merge their functionality? e.g. add a flag

cosicp commented:
> What's the difference between `setSolidFraction` and `compressibleSetSolidFraction`? Do you think we could merge their functionality? e.g. add a flag

Yep, thats what I basically did. I just pushed it as a separate setSolidFraction for now to avoid problems with github actions and so on. I will actually remove the old one now so theres less confusion.

The compressible one can do the same thing as the basic one with the proper flags so it should replace the old one completely. 

You can run it with a flag now, for example:  **compressibleSetSolidFraction -N 8 -compressible -alpha.phase1 "alpha.metal1" -alpha.phase2 "alpha.air"**

The difference is that regular setSolidFraction is built for two phase flow so it sets alpha of one filed (metal) and the solver always knows that the other field is 1 - alpha.metal, but when using more than two phases all of them have to be set at the start. Using the -compressible flag the app sets the alpha.metal  and also sets the volume fractions of alpha.air field (as 1-alpha.metal) and then the vapour phase is 0 at the start everywhere.

cosicp committed:
removing old setSolidFraction

cosicp commented:
I removed the old setSolidFraction and updated it with the new one. Its still the same name but the new functionality can be specified with flags. 

N is 10 by default but by setting -N 1 flag we ge old setSolidFraction. This is what I did in the tutorials to speedup github actions and then the user can set N to something else in the allrun scripts.

-compressible flag tells the app to run in compressible mode so set the alpha.air field also or whatever the name is

I also added the option so the user can specify what is the name of the fields (since they can be different in the compressible solver)  

This can be done with -alpha.phase1 and -alpha.phase2 flags, so for example:

Exmaple 1: 3 phases

**setSolidFraction -N 8 -compressible -alpha.phase1 "alpha.metal1" -alpha.phase2 "alpha.air"**

runs setSolidFraction with N = 8x8x8 in the compressible mode (so it writes out all the fileds) where the first phase is called alpha.metal1 and the second one is alpha.air 


Exmaple 2: Old setSolidFraction for 2 phases where the first phase is called alpha.metal by default

**setSolidFraction -N 1** 

Exmaple 3: 2 phases but N=10 by default

 **setSolidFraction**

Exmaple 4: 3 phases, N=10 by default, alpha.metal and alpha.air default names for phases

**setSolidFraction  -compressible**

philipcardiff commented on applications/utilities/setSolidFraction/setSolidFraction.C:
```diff
@@ -1,84 +1,284 @@
+    \\  /    A nd           | Web: http://www.foam-extend.org
+     \\/     M anipulation  |
 -------------------------------------------------------------------------------
-License
```
Can you set the alignment back to the original/standard alignment for the license and comments?

Separately, we should really replace all references to OpenFOAM/foam with laserbeamFoam (we do this in solids4foam), but let's leave that for a spearate issue - I have a script we can use

philipcardiff commented:
> I removed the old setSolidFraction and updated it with the new one. Its still the same name but the new functionality can be specified with flags.
> 
> N is 10 by default but by setting -N 1 flag we ge old setSolidFraction. This is what I did in the tutorials to speedup github actions and then the user can set N to something else in the allrun scripts.
> 
> -compressible flag tells the app to run in compressible mode so set the alpha.air field also or whatever the name is
> 
> I also added the option so the user can specify what is the name of the fields (since they can be different in the compressible solver)
> 
> This can be done with -alpha.phase1 and -alpha.phase2 flags, so for example:
> 
> Exmaple 1: 3 phases
> 
> **setSolidFraction -N 8 -compressible -alpha.phase1 "alpha.metal1" -alpha.phase2 "alpha.air"**
> 
> runs setSolidFraction with N = 8x8x8 in the compressible mode (so it writes out all the fileds) where the first phase is called alpha.metal1 and the second one is alpha.air
> 
> Exmaple 2: Old setSolidFraction for 2 phases
> 
> **setSolidFraction -N 1**
> 
> Exmaple 3: 2 phases but N=10 by default
> 
> **setSolidFraction**
> 
> Exmaple 4: 3 phases, N=10 by default, alpha.metal and alpha.air default names for phases
> 
> **setSolidFraction -compressible**

Great.

One minor suggestion: can we replace `-N` with something more descriptive, e.g. `-refinement`, or `-subDivisions` or `-nSubCells` or `-nSubdivisions`? Also, for me, `-N` is close to `-n/-np`, which I associate with parallelism.

Apart from that and my one other style comment, I am happy to approve.

cosicp committed:
reverting indentation in the header

cosicp committed:
changing N to nSubCells

cosicp committed:
chaning nSubCells to subDivisions

cosicp commented:
> > I removed the old setSolidFraction and updated it with the new one. Its still the same name but the new functionality can be specified with flags.
> > N is 10 by default but by setting -N 1 flag we ge old setSolidFraction. This is what I did in the tutorials to speedup github actions and then the user can set N to something else in the allrun scripts.
> > -compressible flag tells the app to run in compressible mode so set the alpha.air field also or whatever the name is
> > I also added the option so the user can specify what is the name of the fields (since they can be different in the compressible solver)
> > This can be done with -alpha.phase1 and -alpha.phase2 flags, so for example:
> > Exmaple 1: 3 phases
> > **setSolidFraction -N 8 -compressible -alpha.phase1 "alpha.metal1" -alpha.phase2 "alpha.air"**
> > runs setSolidFraction with N = 8x8x8 in the compressible mode (so it writes out all the fileds) where the first phase is called alpha.metal1 and the second one is alpha.air
> > Exmaple 2: Old setSolidFraction for 2 phases
> > **setSolidFraction -N 1**
> > Exmaple 3: 2 phases but N=10 by default
> > **setSolidFraction**
> > Exmaple 4: 3 phases, N=10 by default, alpha.metal and alpha.air default names for phases
> > **setSolidFraction -compressible**
> 
> Great.
> 
> One minor suggestion: can we replace `-N` with something more descriptive, e.g. `-refinement`, or `-subDivisions` or `-nSubCells` or `-nSubdivisions`? Also, for me, `-N` is close to `-n/-np`, which I associate with parallelism.
> 
> Apart from that and my one other style comment, I am happy to approve.

Sure. I think -subDivisions is the most straightforward one. This is changed now.

philipcardiff reviewed (approved):
All good for me now!

@tomflint22 : you're next!
